packages/create-amplify/src/initial_project_file_generator.ts

```diff
@@ -42,7 +43,7 @@ export class InitialProjectFileGenerator {
       JSON.stringify(packageJsonContent, null, 2)
     );
 
-    if (process.env.PACKAGE_MANAGER_EXECUTABLE === 'yarn-modern') {
+    if (this.packageManagerProps.name === 'yarn-modern') {
```

Comment:
this logic should be moved the PackageManagerController. Something like `packageManagerController.initializeFiles()`

See: https://en.wikipedia.org/wiki/Law_of_Demeter and https://martinfowler.com/bliki/TellDontAsk.html